v2/tools/generator/internal/config/group_configuration.go

```diff
@@ -56,6 +56,24 @@ func (gc *GroupConfiguration) TypeRename(name astmodel.TypeName) (string, error)
 	return rename, nil
 }
 
+// FindUnusedTypeRenames returns a slice listing any unused type rename configuration
+func (gc *GroupConfiguration) FindUnusedTypeRenames() []string {
+	var result []string
+
+	// All our versions are listed twice, under two different keys, so we hedge against processing them multiple times
+	versionsSeen := astmodel.MakeStringSet()
+	for _, vc := range gc.versions {
+		if versionsSeen.Contains(vc.name) {
+			continue
+		}
+
+		versionsSeen.Add(vc.name)
+		result = appendWithPrefix(result, fmt.Sprintf("group %s ", gc.name), vc.FindUnusedTypeRenames()...)
+	}
+
+	return result
+}
```

Comment:
Maybe make an internal method that takes the vc method to call, since the code of this and `FindUnusedARMReferences` is identical? I think you can use something similar to Python's unbound methods...
Yup : https://go.dev/play/p/7JbmJ5H1X3f
So the internal method can just take a `func(*VersionConfiguration) []string` and you can pass `VersionConfiguration.FindUnusedTypeRenames` or `.FindUnusedARMReferences` from the public methods.

Reply:
Agreed - this is worthwhile through the entire hierarchy, as it will simplify future expansions of the configuration for other details. Didn't need the trampoline function from your example though.